modules/op-nonroot-buildah-user-namespaces.adoc

```diff
@@ -0,0 +1,70 @@
+// This module is included in the following assemblies:
+// * secure/unprivileged-building-of-container-images-using-buildah.adoc
+
+:_mod-docs-content-type: PROCEDURE
+
+[id="nonroot-buildah-user-namespaces_{context}"]
+= Running Buildah as a non-root user by configuring user namespaces
+
+Configuring user namespaces is the simplest way to run Buildah in a task as a non-root user. However, some images might not build using this option.
+
+.Prerequisites
+
+* You have installed the `oc` command-line utility.
+
+.Procedure
+
+. Create a copy of the `buildah` task, which is provided in the `openshift-pipelines` namespace; change the name of the copy to `buildah-as-user`.
++
+[source,terminal]
+----
+$ oc get task buildah -n openshift-pipelines -o yaml | yq '. |= (del .metadata |= with_entries(select(.key == "name" )))' | yq '.kind="Task"' | yq '.metadata.name="buildah-as-user"' | oc create -f -
+----
+
+. Edit the copied `buildah` task by running the following command:
++
+[source,terminal]
+----
+$ oc edit task buildah-as-user
+----
++
+In the new task, create `annotations:` and `stepTemplate` sections, as shown in the following example:
++
+[source,yaml]
+----
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  annotations:
+    io.kubernetes.cri-o.userns-mode: 'auto:size=65536;map-to-root=true'
+    io.openshift.builder: 'true'
+  name: assemble-containerimage
+  namespace: pipeline-namespace
+spec:
+  description: This cluster task builds an image.
+#  ...
+  stepTemplate:
+    env:
+      - name: HOME
+        value: /tekton/home
+    image: $(params.builder-image)
+    imagePullPolicy: IfNotPresent
+    name: ''
+    resources:
+      limits:
+        cpu: '1'
+        memory: 4Gi
+      requests:
+        cpu: 100m
+        memory: 2Gi
+    securityContext:
+      capabilities:
+        add:
+          - SETFCAP
+      runAsNonRoot: true
+      runAsUser: 1000 # not strictly necessary since podTemplate is used
+    workingDir: $(workspaces.working-directory.path)
+#  ...
+----
```

Comment:
some ideas for annotations:
```suggestion
----
+
<1> The `annotations` section. <insert what it does and why it's important>
<2> The `stepTemplate` section. <insert what it does and why it's important>
<3> Optional: You can set the `runAsUser` field to have a value of `1000`. If you use a pod template you do not need to include this field.
```

Reply:
I did not add the suggested details because such addition would require a new review cycle white the information is not necessary to complete the task; I did, however, change the inline comment to an annotation.